src/main/java/hudson/plugins/scm/koji/client/KojiBuildDownloader.java

```
@@ -99,7 +97,28 @@ public KojiBuildDownloadResult invoke(File workspace, VirtualChannel channel) th
             targetDir.mkdirs();
         }
         List<String> rpmFiles = downloadRPMs(targetDir, build);
-        
+        String srcUrl = "";
+        for (String suffix : RPM.Suffix.INSTANCE.getSuffixes()) {
+            srcUrl = composeSrcUrl(build.getDownloadUrl(), build, suffix);
+            if (isUrlReachable(srcUrl)) {
+                build.setSrcUrl(new URL(srcUrl));
+            }
+        }
+        // if source file is not found, we try find the directory it might be found in
+        if (build.getSrcUrl() == null) {
+            URL url = new URL(srcUrl);
+            try {
+                do {
+                    URI uri = url.toURI();
+                    // https://stackoverflow.com/questions/10159186/how-to-get-parent-url-in-java
+                    uri = uri.getPath().endsWith("/") ? uri.resolve("..") : uri.resolve(".");
+                    url = uri.toURL();
+                } while (!isUrlReachable(url.toString()));
+            } catch (URISyntaxException e) {
+                e.printStackTrace();
```

are you 100% sure you want just ignore this exception?